src/python/pants/backend/go/target_types.py

```diff
@@ -21,11 +21,38 @@
     StringField,
     Target,
     TargetGenerator,
+    TriBoolField,
     ValidNumbers,
     generate_multiple_sources_field_help_message,
 )
 from pants.util.strutil import softwrap
 
+# -----------------------------------------------------------------------------------------------
+# Build option fieklds
+# -----------------------------------------------------------------------------------------------
+
+
+class GoCgoEnabledField(TriBoolField):
+    """Enables whether Cgo support will be enabled."""
+
+    alias = "cgo_enabled"
+    help = softwrap(
+        """
+        Enable Cgo support, which allows Go and C code to interact. This option must be enabled for any
+        packages making use of Cgo to actually be compiled with Cgo support.
+
+        This field can be specified on several different target types, including `go_binary` and `go_mod` target
+        types. If this field is specified on a `go_binary` target, then that instance takes precedence over other
+        configuration when building the applicable executable. The applicable `go_mod` target will be checked next
+        as a fallback. Finally, if neither target specifies this field, then the value will be taken from
+        the value of the `[golang].cgo_enabled` option. (Note: That option will be deprecated in a future Pants
```

Comment:
So the idea is that `__defaults__` would be the way to have a global setting?

Reply:
That is what @Eric-Arellano was suggesting. The `--go-cgo-enabled` option already exists so this PR still checks its value and does not deprecate it at the moment.

Reply:
But for build options to be added in subsequent PRs, `__defaults_` seems to be the recommended route.

It may be different for tests though where it may be useful to enable certain build options (like the Go data race detector) via a CLI option.

Probably worth opening a GitHub Discussion?

Reply:
I think it's fine as-is, was just wondering about if/when to deprecate.

Reply:
My point is that having CLI options for the default helps with certain build options used when running tests. The data race detector is one such option. So I'm not convinced that `__defaults__` is the whole answer.